cachedcarthage/cache_test.go

```diff
@@ -0,0 +1,262 @@
+}
+
+func Test_GivenCarthageDirDoesNotExist_WhenCreateCalled_ThenExpectCarthageDirAndCacheFileToExist(t *testing.T) {
+	// Given
+	tempDir := givenTempDir(t)
+	expectedDir := filepath.Join(tempDir, "Carthage")
+	expectedFile := filepath.Join(expectedDir, "Cachefile")
+	defer func() {
+		err := os.RemoveAll(tempDir)
+		require.NoError(t, err)
+	}()
+	project := Project{tempDir}
+	mockStateProvider := givenMockProjectStateProvider().GivenParseStateSucceeds(ProjectState{})
+	mockFileCache := givenMockFileCache()
+
+	cache := Cache{
+		project:       project,
+		swiftVersion:  "whatever",
+		filecache:     mockFileCache,
+		stateProvider: mockStateProvider,
+	}
+
+	// When
+	actualError := cache.Create()
+
+	// Then
+	assert.NoError(t, actualError)
+	assert.DirExists(t, expectedDir)
+	assert.FileExists(t, expectedFile)
+}
+
+// cacheFileContent
+func Test_WhenCacheFileContentCalled_ThenExpectCorrectValue(t *testing.T) {
+	// Given
+	resolvedFileName := "Cartfile.resolved"
+	content := "nice content"
+	swiftVersion := "5.0.2"
+
+	expectedContent := fmt.Sprintf("--Swift version: %s --Swift version \n --%s: %s --%s",
+		swiftVersion,
+		resolvedFileName,
+		content,
+		resolvedFileName)
+
+	mockStateProvider := givenMockProjectStateProvider()
+	mockFileCache := givenMockFileCache()
+
+	cache := Cache{
+		project:       Project{},
+		swiftVersion:  swiftVersion,
+		filecache:     mockFileCache,
+		stateProvider: mockStateProvider,
+	}
+
+	// When
+	actualContent := cache.createContentOfCacheFile(content)
+
+	// Then
+	assert.Equal(t, expectedContent, actualContent)
+}
+
+// Collect
+func Test_GivenFileCacheCommitFails_WhenCollectCalled_ThenExpectError(t *testing.T) {
+	// Given
+	expectedError := errors.New("failed to commit cache paths")
+	mockStateProvider := givenMockProjectStateProvider()
+	mockFileCache := givenMockFileCache().
+		GivenIncludeSucceeds().
+		GivenCommitFails(expectedError)
+	cache := Cache{
+		project:       Project{},
+		swiftVersion:  "whatever",
+		filecache:     mockFileCache,
+		stateProvider: mockStateProvider,
+	}
+
+	// When
+	actualError := cache.Collect()
+
+	// Then
+	assert.EqualError(t, expectedError, actualError.Error())
+}
+
+func Test_GivenFileCacheCommitSucceeds_WhenCollectCalled_ThenExpectIncludePathCalledWithCorrectValue(t *testing.T) {
+	// Given
+	projectDir := "/awesomepath"
+	expectedCacheCall := fmt.Sprintf(
+		"%s -> %s",
+		filepath.Join(projectDir, "Carthage"),
+		filepath.Join(projectDir, "Carthage/Cachefile"))
+	mockStateProvider := givenMockProjectStateProvider()
+	mockFileCache := givenMockFileCache().
+		GivenIncludeSucceeds().
+		GivenCommitSucceeds()
+	cache := Cache{
+		project:       Project{projectDir},
+		swiftVersion:  "whatever",
+		filecache:     mockFileCache,
+		stateProvider: mockStateProvider,
+	}
+
+	// When
+	actualError := cache.Collect()
+
+	// Then
+	assert.NoError(t, actualError)
+	mockFileCache.AssertCalled(t, "IncludePath", expectedCacheCall)
```

Comment:
Should we also assert the call to `Commit`?

Reply:
Nice! Yes, we should!